python/ray/serve/_private/router.py

```diff
@@ -495,6 +495,25 @@ async def _resolve_deployment_responses(
         # Return new args and new kwargs
         return new_args, new_kwargs
 
+    def process_finished_request(self, replica_id: ReplicaID, result):
```

Comment:
nit: private method and add type hint
```suggestion
    def _process_finished_request(self, replica_id: ReplicaID, result: Any):
```

please add a docstring indicating this is called by Ray core and what the result will be

Reply:
maybe even mark it as `Union[Any, RayError]` for clarity